### PR TITLE
Auditor accepts arguments, update build and testing tools

### PR DESCRIPTION
**Pull request by ckundo**
- Use webpack and babel to package/transpile.
- Use karma for running tests.
- Use mocha/chai/sinon for testing and assertions.
- Add test for auditor.
- Auditor supports arguments to specify environment and exception behavior

### CODE REVIEW

File: .hound.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+javascript:
+  enabled: false
+jshint:
+  enabled: false
+  config_file: .javascript-style.json
```
Comment (geoffharcourt) [inline .hound.yml]: If we're disabling this (`enabled: false`), do we need to commit the config file?

File: src/index.js [hunk 2]
```diff
@@ -1,3 +1,9 @@
 import auditor from "./auditor.js";
 
-auditor();
+const URL = "https://accesslint-service-staging.herokuapp.com/api/v1/reports";
```
Comment (geoffharcourt) [inline src/index.js]: :+1:

**Comment by geoffharcourt** [comment]
Some of this boilerplate is a bit outside my experience, but LGTM after the question about the Hound config is resolved.

File: bin/setup [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+
+npm install -g webpack gulp-cli
```
Comment (BlakeWilliams) [inline bin/setup]: Thoughts on using npm scripts instead of gulp?

File: src/logger.js [hunk 4]
```diff
@@ -1,5 +1,16 @@
-import request from "request";
+import request from "browser-request";
 
-export default function (message) {
-  request.post(`${window.location}accesslint`, message)
+export default function (message, url, raiseError) {
+  let violations = message.violations;
```
Comment (BlakeWilliams) [inline src/logger.js]: const if you're not rebinding?

File: test/unit/auditor_test.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+"use strict";
+
+let chai = require("chai");
```
Comment (BlakeWilliams) [inline test/unit/auditor_test.js]: Const for these? No ES6?

File: webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    filename: "accesslint.js"
+  },
+  plugins: [],
+  module: {
+    loaders: [
+    {
+      test: /\.js$/,
+      exclude: /node_modules/,
+      loader: "babel-loader",
+      query: {
```
Comment (BlakeWilliams) [inline webpack.config.js]: This gets inherited from .babelrc I believe.

**Pull request merged**